src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -49,19 +54,25 @@
      * @return map {age : people with this age}
      */
     public Map<Integer, List<Person>> groupByAge(final List<Person> people) {
-        return null;
+        return people.stream().collect(groupingBy(Person::getAge));
     }
 
     /**
      * Creates single string, representing all people names, emphasizing uniqueness!
      * Example:
-     *    input:  [{13, "Harry"}, {13, "Ron"}, {14, "Hermione"}, {13, "Harry"}]
-     *    output: "Distinct names: Harry, Ron, Hermione!"
+     * input:  [{13, "Harry"}, {13, "Ron"}, {14, "Hermione"}, {13, "Harry"}]
+     * output: "Distinct names: Harry, Ron, Hermione!"
      *
      * @param people collection of people
      * @return string with unique names, like "Distinct names: a, b, c!"
      */
     public String collectDistinctNames(final List<Person> people) {
-        return null;
+        StringBuilder names = new StringBuilder("Distinct names: ");
+        names.append(people.stream().
+                map(Person::getName).
+                distinct().
```

Comment:
I think there is non-written agreement to keep dot and following method call together. This means prefer
```
people.stream()
      .map(..)
      .collect(..)
```
over
```
people.stream().
      map(..).
      collect(..)
```